packages/api/src/codegen/codegenerator.ts

```diff
@@ -25,7 +25,14 @@ export default abstract class CodeGenerator {
 
   userAgent: string;
 
-  requiredPackages!: Record<string, { reason: string; url: string }>;
+  requiredPackages!: Record<
+    string,
+    {
+      reason: string;
```

Comment:
as a follow-up to the work I added in https://github.com/readmeio/api/pull/759, i was thinking it might make sense to have a `devDep` flag that splits out our required deps (e.g., `@readme/api-core`) from our required dev-deps (e.g., `typescript`)

```suggestion
      devDependency: boolean;
      reason: string;
```

Reply:
will do this in another PR since it's going to require a little bit of work